### PR TITLE
feat: hover effects, abstract common styles, transition timings, brea…

### PR DESCRIPTION
**Pull request by kemuru**
…dcrumb wrap and gap

---

## PR-Codex overview
This PR focuses on enhancing the UI components library by making several stylistic updates, including theme adjustments, hover effects, and component styling improvements across various files.

### Detailed summary
- Updated `<title>` in `src/index.html` to "UI Components Library".
- Added `flex-wrap` and `gap` in `src/lib/breadcrumb.tsx`.
- Changed color variables in `src/lib/accordion/index.tsx`.
- Defined new light and dark theme names in `src/styles/themes.ts`.
- Introduced hover transition styles in `src/styles/common-style.ts`.
- Applied hover effects in multiple components (`datepicker`, `field`, `dropdown`, `file-uploader`, `pagination`, `accordion`, `card`, `tag`, `simple-button`, `progress`) for better user interaction.
- Refactored background color properties to `background-color` for consistency.
- Enhanced the `HorizontalBullet` component to dynamically adjust `paddingBottom` based on content size.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

## Summary by CodeRabbit

- **New Features**
  - Enhanced hover effects and visual behavior for various components, including `AccordionItem`, `DropdownButton`, and `FileUploader`.
  - Added new properties to themes for improved theming support.
  - Introduced new styles for hover effects in the common styles file.

- **Bug Fixes**
  - Improved layout flexibility for the `Breadcrumb` component.

- **Documentation**
  - Updated styles in the common styles file to include new hover transition timings and background colors.

- **Refactor**
  - Updated several components to utilize new styles and improve hover interactions without altering core functionality.
  - Transitioned `HorizontalBullet` to a stateful component for better layout management.

- **Style**
  - Modified the background and hover styles across multiple components for better visual consistency.

### CODE REVIEW

File: src/styles/common-style.ts [hunk 1]
```diff
@@ -113,3 +113,36 @@ export const borderBox = css`
...
 `;
+
+export const hoverShortTransitionTiming = css`
+  :hover {
+    transition: 0.1s;
+  }
+
+  :not(:hover) {
+    transition: 0.1s;
+  }
```
Comment (alcercu) [inline src/styles/common-style.ts:124]: I think this should give the same behavior:
```suggestion
  transition: 0.1s;
```

File: src/styles/common-style.ts [hunk 2]
```diff
@@ -113,3 +113,36 @@ export const borderBox = css`
...
+`;
+
+export const hoverLongTransitionTiming = css`
+  :hover {
+    transition: 0.2s;
+  }
+
+  :not(:hover) {
+    transition: 0.2s;
+  }
```
Comment (alcercu) [inline src/styles/common-style.ts:134]:
```suggestion
  transition: 0.2s;
```

File: src/lib/progress/steps/horizontal-bullet.tsx [hunk 3]
```diff
@@ -93,14 +104,26 @@
...
-);
+}) => {
+  const textRef = useRef<HTMLDivElement>(null);
+  const [paddingBottom, setPaddingBottom] = useState(0);
+
+  useLayoutEffect(() => {
+    if (textRef.current) {
+      const totalHeight = textRef.current.clientHeight;
+      setPaddingBottom(totalHeight);
+    }
```
[the remark this answers is not in the document]
Comment (alcercu) [reply src/lib/progress/steps/horizontal-bullet.tsx:115]: @kemuru Check if you could use the `useElementSize` hook here as pointed out by the rabbit

**Review by alcercu — CHANGES_REQUESTED** [review]

File: src/lib/progress/steps/horizontal-bullet.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (kemuru) [reply src/lib/progress/steps/horizontal-bullet.tsx:115]: yep works great

**Review by alcercu — APPROVED** [review]
lgtm

**Pull request merged**